### PR TITLE
Tidy docs for Device API

### PR DESCRIPTION
**Pull request by sftim**
## Changes

Tidying for https://github.com/firecracker-microvm/firecracker/blob/main/docs/device-api.md

## Reason

Minor cleanups for readability and context.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2]. _not applicable_
- [ ] User-facing changes are mentioned in `CHANGELOG.md`. _not applicable_
- [ ] All added/changed functionality is tested. _not applicable_
- [ ] New `TODO`s link to an issue. _not applicable_
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by ShadowCurse — COMMENTED** [review]
Hi @sftim, thx for the PR. Unfortunately changes here seem not very accurate/correct.
Can you describe why did you think they were necessary?

File: docs/device-api.md [hunk 1]
```diff
@@ -1,14 +1,17 @@
 # Device
 
-The Device-API following functionality matrix indicates which devices are
+Firecracker provides the _Device_ API to let you manage the virtual hardware
+available inside each microVM.
```
[the remark this answers is not in the document]
Comment (sftim) [reply docs/device-api.md:4]: OK. Good to know. The existing doc is confusing.

**Comment by sftim** [comment]
I haven't found time yet to redo this.

**Comment by roypat** [comment]
Hi @sftim, 
are you still interested in working on this?

**Comment by ShadowCurse** [comment]
closing due to inactivity

**Pull request closed without merging**